Game/model/model.gd

```diff
@@ -305,27 +305,39 @@ func get_storage_cap(player_id: int, type: Types.Item) -> float:
 	else:
 		return 0.0
 
+
 ## Set the heat data for the given player to the data given.
 func add_heat_data_at(
-	player_id: int, grid_position: Vector2i, heat: float, heat_capacity: float
+	player_id: int,
+	grid_position: Vector2i,
+	heat: float,
+	heat_capacity: float,
+	heat_state: Types.HeatState
 ) -> void:
-	add_heat_data_for_all_players.rpc(player_id, grid_position, heat, heat_capacity)
+	add_heat_data_for_all_players.rpc(player_id, grid_position, heat, heat_capacity, heat_state)
+
 
 ## Set the heat data for the given player to the data given for all players. Is an RPC.
 @rpc("any_peer", "call_local", "reliable")
 func add_heat_data_for_all_players(
-	player_id: int, grid_position: Vector2i, heat: float, heat_capacity: float
```

Comment:
This is getting *quite* large.  It probably is in need of a refactor, but that's fine for now.